src/workload.rs

```diff
@@ -131,11 +167,14 @@ impl Workload {
             service_account: self.service_account.clone(),
         }
     }
-    pub fn choose_waypoint_address(&self) -> Option<IpAddr> {
-        self.waypoint_addresses
-            .iter()
-            .choose(&mut rand::thread_rng())
-            .copied()
+    pub fn waypoint_svc_ip_address(&self) -> Option<IpAddr> {
```

Comment:
nit:

```rust
        self.waypoint.as_ref().and_then(|gw_address| {
            match &gw_address.destination {
                gatewayaddress::Destination::Hostname(_) => None, // TODO: support this
                gatewayaddress::Destination::Address(ip) => Some(ip.address),
            }
        })
```

is more idiomatic

Reply:
addressed in a slightly different way in https://github.com/istio/ztunnel/pull/486/commits/ecc06af22e028572215f01f205db579f13801728. I opted to return an error here instead of None, so the proposed code here does not work anymore. But I think this is clearer for devs in the meantime until hostname is supported